### PR TITLE
feat: vc wallet - derive credential

### LINKED ISSUE #2694: VC wallet: derive credential feature

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- Closes #2694

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: pkg/wallet/wallet.go [hunk 1]
```diff
@@ -361,7 +361,32 @@ func (c *Wallet) Verify(options VerificationOption) (bool, error) {
...
+	vc, err := c.resolveCredentialToDerive(credential)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve request : %w", err)
+	}
+
+	pkFetcher := verifiable.WithPublicKeyFetcher(
+		verifiable.NewVDRKeyResolver(c.ctx.VDRegistry()).PublicKeyFetcher(),
+	)
+
+	derived, err := vc.GenerateBBSSelectiveDisclosure(options.Frame, []byte(options.Nonce), pkFetcher)
```
Comment (troyronda) [inline pkg/wallet/wallet.go:381]: TODO: we should followup with a refactor of `GenerateBBSSelectiveDisclosure` into a `NewCredential` function and WithSelectiveDisclosure(vc, frame, nonce, ...) option.

**Review by troyronda — APPROVED** [review]

File: pkg/wallet/wallet.go (hunk 1 above)
Comment (sudeshrshetty) [reply pkg/wallet/wallet.go:381]: Will create a task for this

File: pkg/client/vcwallet/client_test.go [hunk 2]
```diff
@@ -134,6 +138,52 @@ const (
...
+            "referenceNumber": 83294847,
+            "type": ["VerifiableCredential", "UniversityDegreeCredential"]
+        }`
+
+	sampleFrame = `
+		{
+			"@context": [
+    			"https://www.w3.org/2018/credentials/v1",
+        		"https://www.w3.org/2018/credentials/examples/v1",
+				"https://w3id.org/security/bbs/v1"
```
[the remark this answers is not in the document]
Comment (sudeshrshetty) [reply pkg/client/vcwallet/client_test.go:169]: fixed

File: pkg/wallet/wallet.go (hunk 1 above)
Comment (sudeshrshetty) [reply pkg/wallet/wallet.go:381]: Created: https://github.com/hyperledger/aries-framework-go/issues/2697

**Pull request merged**